### PR TITLE
Update tokio and other dependencies

### PR DESCRIPTION
**Pull request by thibault-martinez**
## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked that new and existing unit tests pass locally with my changes
- [x] I have updated the CHANGELOG.md, if my changes are significant enough

### CODE REVIEW

File: bee-runtime/Cargo.toml [hunk 1]
```diff
@@ -14,9 +14,9 @@ homepage = "https://www.iota.org"
...
 
 async-trait = "0.1"
-dashmap = "3.11"
+dashmap = "4.0"
 futures = "0.3"
 log = { version = "0.4", features = ["serde"] }
 
 [dev-dependencies]
-tokio = { version = "0.2", features = ["rt-core", "time", "macros"] }
+tokio = { version = "1.0", features = ["rt", "macros", "time"] }
```
Comment (samuel-rufi) [inline bee-runtime/Cargo.toml:22]: do we need `rt-multi-thread` here too?

File: bee-runtime/Cargo.toml [hunk 2]
```diff
@@ -14,9 +14,9 @@ homepage = "https://www.iota.org"
 bee-storage = { version = "0.2.0-alpha", path = "../bee-storage/bee-storage" }
 
 async-trait = "0.1"
-dashmap = "3.11"
+dashmap = "4.0"
```
Comment (samuel-rufi) [inline bee-runtime/Cargo.toml:17]: do we still need dashmap in the node?

File: bee-runtime/Cargo.toml (hunk 1 above)
Comment (thibault-martinez) [reply bee-runtime/Cargo.toml:22]: We don't for the tests

File: bee-runtime/Cargo.toml (hunk 2 above)
Comment (thibault-martinez) [reply bee-runtime/Cargo.toml:17]: Maybe we'll refactor the event bus to get rid of it, but outside of the scope of this PR

**Review by samuel-rufi — APPROVED** [review]

**Comment by Alex6323** [comment]
There are also many lines changed due to rand being updated with this PR. Maybe we should at least mention that.

**Review by Alex6323 — APPROVED** [review]

**Review by zesterer — APPROVED** [review]

**Pull request merged**